src/librustc_driver/lib.rs

```diff
@@ -657,6 +658,29 @@ impl RustcDefaultCalls {
                         }
                     }
                 }
+                PrintRequest::TargetCPUs => {
+                    let tm = create_target_machine(sess);
+                    unsafe { llvm::LLVMRustPrintTargetCPUs(tm); }
+                }
+                PrintRequest::TargetFeatures => {
+                    let tm = create_target_machine(sess);
+                    unsafe { llvm::LLVMRustPrintTargetFeatures(tm); }
+                }
+                PrintRequest::RelocationModels => {
+                    println!("Available relocation models:\n");
```

Comment:
It might be good to make a list of available strings and their enum mappings instead of duplicating them, but not sure where to put it. This information is currently processed in `librustc_trans/back/write.rs` `create_target_machine` function.